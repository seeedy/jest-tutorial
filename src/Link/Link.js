import React, { useState } from "react";

const STATUS = {
    HOVERED: "hovered",
    NORMAL: "normal"
};

export default function Link(props) {
    const [status, setStatus] = useState(STATUS.NORMAL);

    const onMouseEnter = () => {
        return setStatus(STATUS.HOVERED);
    };

    const onMouseLeave = () => {
        return setStatus(STATUS.NORMAL);
    };

    return (
        <a
            className={status}
            href={props.href || "#"}
            onMouseEnter={onMouseEnter}
            onMouseLeave={onMouseLeave}
        >
            {props.children}
        </a>
    );
}
