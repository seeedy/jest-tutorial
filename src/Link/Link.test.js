// Example for snapshot testing

import React from "react";
import Link from "./Link";
import TestRenderer, { act } from "react-test-renderer";

test("Link changing class when hovered", () => {
    const component = TestRenderer.create(
        <Link href="xailabs.com">Xailabs</Link>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger
    act(() => tree.props.onMouseEnter());
    // re-render
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // trigger
    act(() => tree.props.onMouseLeave());
    // render
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
