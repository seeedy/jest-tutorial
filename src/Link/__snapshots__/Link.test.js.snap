// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changing class when hovered 1`] = `
<a
  className="normal"
  href="xailabs.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Xailabs
</a>
`;

exports[`Link changing class when hovered 2`] = `
<a
  className="hovered"
  href="xailabs.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Xailabs
</a>
`;

exports[`Link changing class when hovered 3`] = `
<a
  className="normal"
  href="xailabs.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Xailabs
</a>
`;
